File: solutions/cpp/two-fer/1/two_fer.cpp
#include "two_fer.h"
#include <string>

namespace two_fer {
    std::string prefix{"One for "};
    std::string suffix{", one for me."};
    
    std::string two_fer(std::string name) {
        if (name.empty()) {
            return prefix + "you" + suffix;
        } else {
            return prefix + name + suffix;
        }
    }
}  

File: solutions/go/two-fer/1/two_fer.go
// This is a "stub" file.  It's a little start on your solution.
// It's not a complete solution though; you have to write some code.

// Package twofer should have a package comment that summarizes what it's about.
// https://golang.org/doc/effective_go.html#commentary
package twofer

// ShareWith returns a string whose value changes if the name is empty or not.
func ShareWith(name string) string {

	s := ""

	if name == "" {
		s = "One for you, one for me."
	} else {
		s = "One for " + name + ", one for me."
	}

	return s
}
